core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/security/CarbonJMXAuthenticator.java

```diff
@@ -137,15 +141,27 @@ public Subject authenticate(Object credentials) {
         }
     }
 
-    private boolean authorize(String userName) throws UserStoreException {
-
-        AuthorizationManager authorizationManager = userRealm.getAuthorizationManager();
 
-        if (authorizationManager != null) {
-            return authorizationManager.isUserAuthorized(userName, JMX_USER_PERMISSION, "ui.execute");
+    private boolean authorize(String userName) throws UserStoreException {
+        UserStoreManager authorizer;
+        boolean isAuthorized = false;
+        try {
+            authorizer = userRealm.getUserStoreManager();
+        } catch (UserStoreException e) {
+            String msg = "Cannot get authorizer from Realm";
+            log.error(msg, e);
+            throw new SecurityException(msg, e);
         }
 
-        throw new UserStoreException("Unable to retrieve Authorization manager to perform authorization");
+        List<String> userRoleslist = Arrays.asList(authorizer.getRoleListOfUser(userName));
+        if(userRoleslist.contains("JMXMonitorRole")){
+            roleName = "JMXMonitorRole";
```

Comment:
Assigning a class variable in a method and using it in another method is not a good practice. There might me race conditions. Check on a possibility avoid that.